### ISSUE TITLE
doc_independent jobs have started failing due to dependencies moving forward to python 3.6 syntax on xenial base

### ISSUE THREAD

**Issue by tfoote** [ros-infrastructure/ros_buildfarm#877]
It started failing 3 days ago. 

https://build.ros.org/job/doc_independent-packages/722/console

```
03:43:26 sphinx-build -b html -d _build/doctrees   . _build/html
03:43:26 Traceback (most recent call last):
03:43:26   File "/usr/local/bin/sphinx-build", line 7, in <module>
03:43:26     from sphinx.cmd.build import main
03:43:26   File "/usr/local/lib/python3.5/dist-packages/sphinx/cmd/build.py", line 23, in <module>
03:43:26     import sphinx.locale
03:43:26   File "/usr/local/lib/python3.5/dist-packages/sphinx/locale/__init__.py", line 106
03:43:26     translators: Dict[Tuple[str, str], NullTranslations] = defaultdict(NullTranslations)
03:43:26                ^
03:43:26 SyntaxError: invalid syntax
03:43:26 Makefile:40: recipe for target 'html' failed
```

In that job it moved from 3.5.4 to 4.0.1  https://pypi.org/project/Sphinx/#history

Pinning sphinx to 3.5.4 gets further https://build.ros.org/job/doc_independent-packages/725/console

```
15:14:11 sphinx-build -b html -d _build/doctrees   . _build/html
15:14:11 Traceback (most recent call last):
15:14:11   File "/usr/local/bin/sphinx-build", line 7, in <module>
15:14:11     from sphinx.cmd.build import main
15:14:11   File "/usr/local/lib/python3.5/dist-packages/sphinx/cmd/build.py", line 25, in <module>
15:14:11     from sphinx.application import Sphinx
15:14:11   File "/usr/local/lib/python3.5/dist-packages/sphinx/application.py", line 32, in <module>
15:14:11     from sphinx.config import Config
15:14:11   File "/usr/local/lib/python3.5/dist-packages/sphinx/config.py", line 27, in <module>
15:14:11     from sphinx.util.tags import Tags
15:14:11   File "/usr/local/lib/python3.5/dist-packages/sphinx/util/tags.py", line 11, in <module>
15:14:11     from jinja2 import nodes
15:14:11   File "/usr/local/lib/python3.5/dist-packages/jinja2/__init__.py", line 5, in <module>
15:14:11     from .bccache import BytecodeCache
15:14:11   File "/usr/local/lib/python3.5/dist-packages/jinja2/bccache.py", line 61
15:14:11     self.code: t.Optional[CodeType] = None
15:14:11              ^
```

Jinja has also jumped up to 3.x from 2.x in the last few hours... https://pypi.org/project/Jinja2/#history

pinning jinja==2.11.3

And markupsafe also jumped forward: https://pypi.org/project/MarkupSafe/#history

```
15:59:41 sphinx-build -b html -d _build/doctrees   . _build/html
15:59:41 Traceback (most recent call last):
15:59:41   File "/usr/local/bin/sphinx-build", line 7, in <module>
15:59:41     from sphinx.cmd.build import main
15:59:41   File "/usr/local/lib/python3.5/dist-packages/sphinx/cmd/build.py", line 25, in <module>
15:59:41     from sphinx.application import Sphinx
15:59:41   File "/usr/local/lib/python3.5/dist-packages/sphinx/application.py", line 32, in <module>
15:59:41     from sphinx.config import Config
15:59:41   File "/usr/local/lib/python3.5/dist-packages/sphinx/config.py", line 27, in <module>
15:59:41     from sphinx.util.tags import Tags
15:59:41   File "/usr/local/lib/python3.5/dist-packages/sphinx/util/tags.py", line 11, in <module>
15:59:41     from jinja2 import nodes
15:59:41   File "/usr/local/lib/python3.5/dist-packages/jinja2/__init__.py", line 6, in <module>
15:59:41     from markupsafe import escape
15:59:41 ImportError: cannot import name 'escape'
```

pinning to markupsafe==1.1.1

That got it to turn over. https://build.ros.org/job/doc_independent-packages/728/console

The problem is that this is still running xenial and that's officially EOL which is why everyone's moving forward.

**Comment by tfoote** [comment]
I believe that this has recurred: https://build.ros.org/job/doc_independent-packages/

It last ran on July 3rd: https://build.ros.org/job/doc_independent-packages/781/

The latest run: https://build.ros.org/job/doc_independent-packages/803/console

Failed with this error
```
03:44:21 sphinx-build -b html -d _build/doctrees   . _build/html
03:44:21 Running Sphinx v3.5.4
03:44:22 
03:44:22 Exception occurred:
03:44:22   File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/doctest.py", line 24, in <module>
03:44:22     from packaging.specifiers import InvalidSpecifier, SpecifierSet
03:44:22   File "/usr/local/lib/python3.5/dist-packages/packaging/specifiers.py", line 99
03:44:22     _operators: Dict[str, str] = {}
03:44:22               ^
03:44:22 SyntaxError: invalid syntax
03:44:22 The full traceback has been saved in /tmp/sphinx-err-111jnlg9.log, if you want to report the issue to the developers.
03:44:22 Please also report this if it was a user error, so that a better error message can be provided next time.
03:44:22 A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
03:44:22 Makefile:40: recipe for target 'html' failed
03:44:22 make: *** [html] Error 2
03:44:22 Build step 'Execute shell' marked build as failure
```

**Comment by nuclearsandwich** [comment]
Fixed by #884.
